pkgs/development/embedded/avrdude/default.nix

```diff
@@ -9,16 +9,22 @@ stdenv.mkDerivation rec {
   pname = "avrdude";
   version = "7.0";
 
-  src = fetchurl {
-    url = "mirror://savannah/${pname}/${pname}-${version}.tar.gz";
-    sha256 = "sha256-wO9l2Y1gQMoLTytwDVFGPCoflGZUQfOdFdl0Qtu3m1Q=";
+  src = fetchFromGitHub {
+    owner = "avrdudes";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-T8MKrvBvFF3WFwBMIN75vCOS0khliHQI+GGQvCk7T1o=";
   };
 
-  configureFlags = lib.optionals docSupport "--enable-doc";
+  nativeBuildInputs = [ cmake bison flex ];
 
-  buildInputs = [ bison flex libusb-compat-0_1 libelf libftdi1 readline ]
+  buildInputs = [ libusb-compat-0_1 libelf libftdi1 readline ]
     ++ lib.optionals docSupport [ texLive texinfo texi2html ];
 
+  cmakeFlags = lib.optionals docSupport [
+    "-DBUILD_DOC=ON"
+  ];
+
```

Comment:
Hm, there's several logical changes going on here, I think. What is the part that's fixing the build for aarrch64-darwin? And what is "nice to have"? (Can you split logical changes like that into separate commits?)

Or, if everything is one logical change, perhaps you could write a sentence in the commit message (body) about how this fixes things?

Reply:
There's nothing else than switching to CMake, so I modified the commit message to this.